includes/http/WP_AI_Client_Client_Adapter.php

```diff
@@ -0,0 +1,192 @@
+		$args = array(
+			'method'      => $request->getMethod(),
+			'headers'     => $this->prepare_headers( $request ),
+			'body'        => $this->prepare_body( $request ),
+			'timeout'     => 30,
+			'redirection' => 5,
+			'httpversion' => $request->getProtocolVersion(),
+			'blocking'    => true,
+		);
+
+		// Handle streaming requests if needed.
+		if ( $request->hasHeader( 'X-Stream' ) ) {
+			$args['stream']   = true;
+			$args['filename'] = $request->getHeaderLine( 'X-Stream-Filename' );
+		}
+
+		return $args;
+	}
+
+	/**
+	 * Prepare headers for WordPress HTTP API.
+	 *
+	 * @param RequestInterface $request The PSR-7 request.
+	 *
+	 * @return array<string, string> Headers array for WordPress HTTP API.
+	 */
+	private function prepare_headers( RequestInterface $request ): array {
+		$headers = array();
+
+		foreach ( $request->getHeaders() as $name => $values ) {
+			// Skip pseudo headers used for streaming.
+			if ( strpos( $name, 'X-Stream' ) === 0 ) {
+				continue;
+			}
+
+			// WordPress expects headers as name => value pairs.
+			$headers[ (string) $name ] = implode( ', ', $values );
+		}
+
+		return $headers;
+	}
+
+	/**
+	 * Prepare request body for WordPress HTTP API.
+	 *
+	 * @param RequestInterface $request The PSR-7 request.
+	 *
+	 * @return string|null The request body.
+	 */
+	private function prepare_body( RequestInterface $request ): ?string {
+		$body = $request->getBody();
+
+		if ( $body->getSize() === 0 ) {
+			return null;
+		}
+
+		// Rewind the stream to ensure we read from the beginning.
+		if ( $body->isSeekable() ) {
+			$body->rewind();
+		}
+
+		return (string) $body;
+	}
+
+	/**
+	 * Create PSR-7 response from WordPress HTTP response.
+	 *
+	 * @param array{headers: \Traversable<string, string|array<string>>|array<string, string|array<string>>, body: string, response: array{code: int|string, message: string}} $wp_response WordPress HTTP API response array.
+	 *
+	 * @return ResponseInterface PSR-7 response.
+	 */
+	private function create_psr_response( array $wp_response ): ResponseInterface {
+		$status_code   = \wp_remote_retrieve_response_code( $wp_response );
+		$reason_phrase = \wp_remote_retrieve_response_message( $wp_response );
+		$headers       = \wp_remote_retrieve_headers( $wp_response );
+		$body          = \wp_remote_retrieve_body( $wp_response );
+
+		// Create the PSR-7 response.
+		$response = $this->response_factory->createResponse( (int) $status_code, $reason_phrase );
+
+		// Add headers to response.
+		if ( $headers instanceof \WP_HTTP_Requests_Response ) {
+			$headers = $headers->get_headers();
+		}
+
+		/**
+		 * Headers from WordPress response.
+		 *
+		 * @var \Traversable<string, string|array<string>>|array<string, string|array<string>> $headers
+		 */
+		if ( is_array( $headers ) || $headers instanceof \Traversable ) {
+			foreach ( $headers as $name => $value ) {
+				// PSR-7 expects string name and string|array value.
+				$response = $response->withHeader( $name, $value );
```

Comment:
Will this append in the case of multiple headers of the same name? Just want to make sure this doesn't accidentally overwrite and maintain only the last one.

Reply:
I'm not sure what intended behavior you're intending. This is simply setting the headers based on the headers in the WP response. Since this is using the PSR-7 interface, it's technically up to the concrete response class as to how it handles this. According to the HTTP spec multiple headers with the same name are supported, so it _should_ work that way.